libs/vkd3d/command.c

```diff
@@ -10799,6 +10877,14 @@ static HRESULT STDMETHODCALLTYPE d3d12_command_queue_Wait(ID3D12CommandQueue *if
     sub.type = VKD3D_SUBMISSION_WAIT;
     sub.wait.fence = fence;
     sub.wait.value = value;
+    sub.wait.handle = NULL;
+
+    if (fence->is_shared)
+    {
+        sub.wait.handle = CreateEventW(NULL, TRUE, FALSE, NULL);
+        shared_fence_wait(fence->shared.fence, sub.wait.handle, value);
```

Comment:
shared_fence_wait() implies that this is a blocking wait. Might be clearer if it's called `fence_add_waiter` or something.